subsys/modbus/modbus_serial.c

```diff
@@ -337,6 +337,11 @@ static void cb_handler_rx(struct modbus_context *ctx)
 		}
 
 	} else {
+		if (cfg->uart_buf_ctr == CONFIG_MODBUS_BUFFER_SIZE) {
+			/* Buffer full. Disable interrupt until timeout. */
+			modbus_serial_rx_disable(ctx);
+			return;
+		}
```

Comment:
```c

		int i;

		if (cfg->uart_buf_ctr == CONFIG_MODBUS_BUFFER_SIZE) {
			/* Buffer full. Disable interrupt until timeout. */
			modbus_serial_rx_disable(ctx);
			return;
		}
```
Please place `int n` at the top of this block.

Btw, on second thought, would it not make sense to move that check to the bottom of that else block?

Reply:
If we move it to the bottom, the timer will be restarted. I think it is better to avoid that, since we want to process the buffer as soon as possible.